src/transport/SecureSession.h

```diff
@@ -209,17 +221,45 @@ class SecureSession : public Session
     SessionMessageCounter & GetSessionMessageCounter() { return mSessionMessageCounter; }
 
 private:
-    Type mSecureSessionType;
+    enum class State : uint8_t
+    {
+        // kPending denotes a secure session object that is internally
+        // reserved by the stack before and during session establishment.
+        //
+        // Although the stack can tolerate eviction of these (releasing one
+        // out from under the holder would exhibit as CHIP_ERROR_INCORRECT_STATE
+        // during CASE or PASE), intent is that we should not and would leave
+        // these untouched until CASE or PASE complete.
+        //
+        // During this stage, the reference counter is hold by the PairingSession
```

Comment:
Why do we need this slightly different behavior where the PairingSession is the one responsible for incrementing the counter?

Why can't we just have the same behavior where upon construction of a `SecureSession` object, even before activation, that it would have a reference count of 1?

Reply:
Resolved offline. PairingSession initially 'owns' the session (and uses SessionHolder's refcounting to achieve this) and then later, transfers the ownership to SessionManager. That is not apparent in the comments here. Will spend some time updating all the comments once this PR merges.